src/ScottPlot/Config/ColorMaps/ColormapFromByteArray.cs

```diff
@@ -21,9 +22,9 @@ public abstract class ColormapFromByteArray : Colormap
             return output;
         }
 
-        public override int[] IntensitiesToARGB(double[] intensities)
+        public override int[] IntensitiesToARGB(double[] intensities, double? transparencyThreshold = null)
         {
-            return intensities.AsParallel().AsOrdered().Select(i => RGBToARGB(new byte[] { cmap[(int)(i * 255), 0], cmap[(int)(i * 255), 1], cmap[(int)(i * 255), 2] })).ToArray();
+            return intensities.AsParallel().AsOrdered().Select(i => i < (transparencyThreshold ?? double.NegativeInfinity) ? unchecked((int)0x00000000) : RGBToARGB(new byte[] { cmap[(int)(i * 255), 0], cmap[(int)(i * 255), 1], cmap[(int)(i * 255), 2] })).ToArray();
```

Comment:
I'm a bit confused with this one 🤔 why is the 0 unchecked?

```cs
return intensities
    .AsParallel()
    .AsOrdered()
    .Select(i => i < (transparencyThreshold ?? double.NegativeInfinity) ? 
        unchecked((int)0x00000000) : 
        RGBToARGB(new byte[] { /* cmap RGB values */ }))
    .ToArray();
```

Reply:
I don't know. It should just be this:
```cs
Select(i => i < (transparencyThreshold ?? double.NegativeInfinity) ? 
        unchecked((int)0x00000000) : 
        RGBToARGB(new byte[] { /* cmap RGB values */ }))
```

I suspect that I copied this without thinking from somewhere which used a constant like this: `0xFFxxxxxx`, and the compiler treats that constant as a long because the largest value an int can hold is `0x7FFFFFFF` because of the sign bit. The unchecked was necessary because it would overflow (intentionally, as ARGBs are unsigned despite being represented as signed ints in .NET). I solved that problem by just writing (0xFF << 24), as bitshifts are not caught by compile-time checking.

So, short story, let the strange code die.

Reply:
Sounds good! It made its way into a few files but I'll update this during the colormap refactor 👍 

Thanks for all your work on this! These heatmaps are looking awesome